### ISSUE TITLE
Sensor on compound head shows 25 degrees cold offset

### ISSUE THREAD

**Issue by sanslos** [PrintM3D/Promega#29]
The idle (cold) temperature on the Compound head shows around 0C.  It heats and cools, but starts and stops around 0C.
![cold 0c to 200c](https://user-images.githubusercontent.com/43072647/47086786-4c19e100-d1cf-11e8-8cd6-fb223c854b34.PNG)
![cooling down from 200c at 21c](https://user-images.githubusercontent.com/43072647/47086798-51772b80-d1cf-11e8-8c18-41a1b2e63d44.PNG)
![cooling down from 20c to 4c](https://user-images.githubusercontent.com/43072647/47086805-53d98580-d1cf-11e8-86b0-b3cd0f00a7fa.PNG)

**Comment by sohjsolwin** [comment]
what version of the duet firmware are you on? Have you updated it from what the printer originally shipped with? There was a change in one of the beta versions that modified the temperature readings a bit. I believe that may be related to what you're seeing (only if you've updated the firmware though).

**Comment by sanslos** [comment]
It's a little more complicated than that.  This problem (low temp on Compound) was first seen with the original firmware.  I upgraded the firmware (currently 2.02RC3), no change, re-calibrated, no change.  I have a similar problem with K'Tana, where initially both sensors read about the same (around 27C at idle), then one sensor suddenly dropped to showing about 12C at idle (no change in firmware).  Note that I suspect a firmware issue, but I put a new sensor in the Compound head and it showed 27C again.  Something weird is going on.
![promega firmware version k tana temp issue](https://user-images.githubusercontent.com/43072647/47858193-b4a8b680-dda8-11e8-83de-fe3616ff2818.PNG)

![k tana initial bringup temps](https://user-images.githubusercontent.com/43072647/47858
![k tana temps t1 low](https://user-images.githubusercontent.com/43072647/47858191-b07c9900-dda8-11e8-8640-d910629606ca.PNG)
183-aa86b800-dda8-11e8-8dea-90e085757903.png)

**Comment by sanslos** [comment]
(Sorry, thought this posted)  This is what the K'Tana shows when first turned on

![k tana initial bringup temps](https://user-images.githubusercontent.com/43072647/47858223-c5f1c300-dda8-11e8-8a27-4300e37635f7.png)

**Comment by sanslos** [comment]
The lower temperature is consistent, it doesn't bounce around.  T1 now always shows 12C when idle.

**Comment by sohjsolwin** [comment]
definitely strange. Have you run a PID tuning yet? https://promega.printm3d.com/repair-and-maintenance/temperature-calibration
Note: You shouldn't normally need to, but with the readings seemingly functioning, just being off by a consistent amount, recalibrating would be my best suggestion.

**Comment by sanslos** [comment]
Yes, I have (for the Compound).  I haven't done it for the current problem, I was planning to, but that wouldn't explain why it started out with both sensors showing 27C then one suddenly dropping to 12C.

**Comment by sanslos** [comment]
The problems I'm seeing are due to the sensors.  I did experiments with the four sensors (Compound, 2 from K'Tana, one replacement) and two Duet ports, I currently have two failed sensors that show low.  Note the original Compound sensor now shows -18C, they deteriorate over time:

                          Duet Port (C)
Sensor   Ohms     Left    Right
        1    1046    12.40    12.00
        2    1106    27.90    27.70
        3    926    -18.10    -18.30
        4    1107    28.00    28.00

        Note: Wire Resistance = 0.6 ohms

**Comment by c-schmitz** [comment]
I can confirm similar issues. Both my sensors are off by -15°C now. In the beginning they worked normally. 
This caused the head being too hot at nominal temperature.

I was able to temporarily fix that by messing with the R value in the configuration However this seems to be a general quality problem with the sensors (like so many other things)

**Comment by c-schmitz** [comment]
Why was this closed?

**Comment by sanslos** [comment]
The issue was with the sensors, when they were replaced the issue was fixed.  I should have closed this when that was verified (last November, 2018), it was the point of my Nov. 6, 2018 post saying "The problems I'm seeing are due to the sensors", although I may not have been clear.

Note that at the time I was also having problems with the heating, [I posted a whole series of experiments on Discord](https://discordapp.com/channels/279255051036524544/461926470776782848/509255194782072852).  Once I replace the sensors and the suspect heaters everything worked fine.

For completeness I should mention that at least once (twice?) I was not able to keep temperature, which was a Duet issue.  A hard power cycle (unplugging the printer) solved that.
